src/Symfony/Component/Config/Definition/Loader/DefinitionFileLoader.php

```diff
@@ -46,9 +46,19 @@ public function load(mixed $resource, ?string $type = null): mixed
         // the closure forbids access to the private scope in the included file
         $load = \Closure::bind(static function ($file) use ($loader) {
             return include $file;
-        }, null, ProtectedDefinitionFileLoader::class);
+        }, null, null);
 
-        $callback = $load($path);
+        try {
+            $callback = $load($path);
+        } catch (\Error) {
+            $load = \Closure::bind(static function ($file) use ($loader) {
+                return include $file;
+            }, $this, ProtectedDefinitionFileLoader::class);
+
+            $callback = $load($path);
+
+            trigger_deprecation('symfony/config', '7.4', 'Accessing the internal scope of the loader in config files is deprecated, use only its public API instead in "%s".', $path);
```

Comment:
Should this include the exact location of the caught error to make it easier to identify the issue ?
In particular, cases using `$this` need to be migrated to `$loader`, which is not mentioned in the message.

Reply:
The deprecation message used in the DI component is much better in this regard.

Reply:
I figured out that `$this` is a concern only for DI: routing and config already use static closures.